### PR TITLE
fix(framework-core): combine review comments

### PR DESCRIPTION
**Pull request by TomKristie**
Collapses timeline and inline comment into a single pull request. This also prompted some refactoring because of the sequence of events change.

Addresses [review comment](https://github.com/googleapis/code-suggester/pull/105#discussion_r485085505)

Towards #105

### CODE REVIEW

File: src/github-handler/comment-handler/make-review-handler/upload-comments-handler.ts [hunk 1]
```diff
@@ -99,15 +97,19 @@ export async function makeInlineSuggestions(
   ).data.head.sha;
   const comments = buildReviewComments(suggestions);
   if (!comments.length) {
-    logger.info('No suggestions to make. Exiting...');
-    return;
+    logger.info('No suggestions to make');
+  }
+  const errorMessage = buildErrorMessage(outOfScopeSuggestions);
+  if (errorMessage) {
+    logger.warn('Some suggestions could not be made');
```
Comment (chingor13) [inline src/github-handler/comment-handler/make-review-handler/upload-comments-handler.ts:104]: I think we're missing the case of no suggestion and no errors (which should do nothing).

**Review by chingor13 — CHANGES_REQUESTED** [review]

File: src/github-handler/comment-handler/make-review-handler/upload-comments-handler.ts (hunk 1 above)
Comment (TomKristie) [reply src/github-handler/comment-handler/make-review-handler/upload-comments-handler.ts:104]: Hm wouldn't it be the case: when we have no suggestions we would still want to comment that out of scope suggestions were generated?

Comment (TomKristie) [reply src/github-handler/comment-handler/make-review-handler/upload-comments-handler.ts:104]: I propose we should do nothing if we have both no suggestions to make and no out of scope suggestions generated (assuming users care about all suggestions generated). Do you agree?

Comment (chingor13) [reply src/github-handler/comment-handler/make-review-handler/upload-comments-handler.ts]: Correct, if there are no suggestions and no errors, we should do nothing.

**Review by chingor13 — APPROVED** [review]

**Pull request merged**